src/pages/Fox/components/FoxHeader.tsx

```diff
@@ -31,6 +31,14 @@ export const FoxHeader = () => {
     history.push('/explore')
   }, [history])
 
+  // @TODO: this doesn't work
+  useEffect(() => {
+    if (!accountIds.length) setAssetAccountId(undefined)
+    if (accountIds.length === 1) {
+      setAssetAccountId(accountIds[0])
+    }
+  }, [accountIds, setAssetAccountId])
+
```

Comment:
I did try to reintroduce this because from the Context, `assetAccountId` is not reset if you did set it from native wallet, you can reproduce by:

- using native wallet
- Switching to metamask
- notice the staking balance is not updated (because assetAccountId is using previously selected account id)